PurchasesTests/Misc/AtomicTests.swift

```diff
@@ -44,4 +44,18 @@ class AtomicTests: XCTestCase {
 
         expect(result) == 20
     }
+
+    func testModifyCanBeReEntrant() {
+        func add(_ atomic: Atomic<Int>, _ number: Int) {
+            atomic.modify { $0 += number }
+        }
+
+        let atomic = Atomic(10)
+        atomic.modify { _ in
+            add(atomic, 10)
```

Comment:
This test fails because of `inout` memory safety, regardless of the lock: https://docs.swift.org/swift-book/LanguageGuide/MemorySafety.html
Will have to update the docs to reflect it.